docs/tutorials/leaderboards.md

```diff
@@ -177,6 +177,90 @@ The message is just a basic message to inform you to the status the download; wh
 - **ugc_handle:** handle for any UGC that is attached to this entry
 - **details:** any details you stored with this entry for later use
 
+{==
+## Uploading UGC to Leaderboard
+==}
+
+You can use this method if you want to attach replays or any other data to leadearboard entry.
+So first you will need to setup signals
+
+=== "Godot 2.x, 3.x"
+
+	```gdscript
+	Steam.connect("file_share_result", self, "_on_file_share_result")
+	Steam.connect("file_share_result", self, "on_file_write_async_complete")
+	Steam.connect("leaderboard_ugc_set", self, "on_leaderboard_ugc_set")
+	```
+
+=== "Godot 4.x"
+
+	```
+	Steam.connect("file_share_result", Callable(self, "on_file_share_result"))
+	Steam.connect("file_write_async_complete", Callable(self, "on_file_write_async_complete"))
+	Steam.connect("leaderboard_ugc_set", Callable(self, "on_leaderboard_ugc_set"))
+	```
+	
+### Creating a file we want to share
+
+Created file is stored in PackedByteArray format.
+Frist we need to create a file with content that we want to share.
+
+```gdscript
+var data:PackedByteArray
+Steam.fileWriteAsync("file_name",data,data.size())
+```
+
+!!! warning "File location"
+	File will be created in location `...\Steam\userdata\<localid?>\<appid>\remote`,
+
+
+Onece it's created you can mark file to be shared
+
+```gdscript
+func on_file_write_async_complete(result):
+	print(result)
+	if result == 1:
+		Steam.fileShare("file_name")
+	elese:
+		pass #Handle errors here
+```
+
+### Understanding FileShare method
+
+Method is used to mark existing files to be shared that are present in remote local dir.
+As argument we are using filename.
+
+```gdscript
+Steam.fileShare("file_name")
+```
+
+!!! warning "File location"
+	Files you want to share must exist in remote location `...\Steam\userdata\<localid?>\<appid>\remote`. Providing absolute path will not work.
+	Method is only marking files to be shared, othere wise if they not exist you will get error, hence in first step we created the file.
```

Comment:
This should be: "The function is only marking files to be shared; otherwise, if they do not exist, you will get error."